Apromore-Custom-Plugins/Process-Simulation-Info-Logic/src/main/java/org/apromore/processsimulation/service/SimulationInfoService.java

```diff
@@ -107,10 +109,12 @@ public ProcessSimulationInfo deriveSimulationInfo(
 
                 deriveGeneralInfo(builder, logSummary);
 
-                deriveTaskInfo(builder, abstraction);
+                deriveTaskInfo(builder, (AbstractAbstraction) abstraction);
 
                 deriveTimetable(builder);
 
+                deriveResourceInfo(builder, (AbstractAbstraction) abstraction);
```

Comment:
How about adding `final AbstractAbstraction abstractAbstraction = (AbstractAbstraction) abstraction;` right at the beginning of the condition block where you test that `abstraction instanceof AbstractAbstraction` and avoid having to cast `abstraction` repeatedly thereafter?

Reply:
Done.